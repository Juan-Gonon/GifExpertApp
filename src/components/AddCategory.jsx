import { useState } from 'react'

// eslint-disable-next-line react/prop-types
export function AddCategory ({ setCategory }) {
  const [inputValue, setInputValue] = useState('')

  const handleChangeInput = (e) => {
    setInputValue(e.target.value)
  }

  const handleSubmitForm = (e) => {
    e.preventDefault()

    if (inputValue.trim().length <= 1) return

    setCategory({ inputValue })
    setInputValue('')
  }

  return (
    <>
      <form action='' onSubmit={handleSubmitForm}>
        <input
          type='text'
          placeholder='Buscar gif...'
          value={inputValue}
          onChange={handleChangeInput}
        />
        <button type='submit'>Buscar...</button>
      </form>
    </>
  )
}
