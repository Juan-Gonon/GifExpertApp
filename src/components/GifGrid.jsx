// eslint-disable-next-line react/prop-types
export function GifGrid ({ category }) {
  return (
    <>
      <h4>{category}</h4>
      <p>Hola Mundo</p>
    </>
  )
}
