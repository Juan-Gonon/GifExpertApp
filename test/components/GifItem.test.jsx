import { describe, it, expect, afterEach } from 'vitest'
import { render, screen, cleanup } from '@testing-library/react'
import { GifItem } from '../../src/components/GifItem'

describe('<GifItem />', () => {
  afterEach(() => {
    cleanup()
  })

  const title = 'Kamila'
  const url = 'https://media3.giphy.com/media/pCGyLbTeliIwqVU9Md/giphy.gif?cid=1470fdd1rxiigktcrpiz7quitx6s9znwhfx50bj355m95trp&ep=v1_gifs_search&rid=giphy.gif&ct=g'

  it('show to give <GifItem />', () => {
    const { container } = render(<GifItem />)
    expect(container).toMatchSnapshot()
  })

  it('debe de mostrar la imagen con el URL y el ALT indicado', () => {
    render(<GifItem title={title} url={url} />)
    screen.debug()

    // expect(screen.getByRole('img').src).toBe(url)
    // expect(screen.getByRole('img').alt).toBe(title)

    const { src, alt } = screen.getByRole('img')
    expect(src).toBe(url)
    expect(alt).toBe(title)
  })

  it('debe de mostrar el titulo en el componente ', () => {
    render(<GifItem title={title} url={url} />)
    expect(screen.getByText(title)).toBeTruthy()
  })
})
