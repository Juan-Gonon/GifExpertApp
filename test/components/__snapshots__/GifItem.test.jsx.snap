// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<GifItem /> > show to give <GifItem /> 1`] = `
<div>
  <div
    class="card"
  >
    <img />
    <p />
  </div>
</div>
`;
